wallet/src/de/schildbach/wallet/ui/SearchUserActivity.kt

```diff
@@ -0,0 +1,210 @@
+                    }
+
+                    override fun onTransitionResume(transition: Transition) {
+                    }
+
+                    override fun onTransitionPause(transition: Transition) {
+                    }
+
+                    override fun onTransitionCancel(transition: Transition) {
+                    }
+
+                    override fun onTransitionStart(transition: Transition) {
+                        layout_title.visibility = View.GONE
+                        find_a_user_label.visibility = View.GONE
+                    }
+
+                })
+                TransitionManager.beginDelayedTransition(root, transition)
+                constraintSet2.applyTo(root)
+                setChanged = true
+            }
+        }
+    }
+
+    private fun initViewModel() {
+        dashPayViewModel = ViewModelProvider(this).get(DashPayViewModel::class.java)
+        dashPayViewModel.searchUsernamesLiveData.observe(this, Observer {
+            if (Status.LOADING == it.status) {
+                startLoading()
+            } else {
+                stopLoading()
+                if (it.data != null) {
+                    adapter.results = it.data
+                    if (it.data.isEmpty()) {
+                        showEmptyResult()
+                    } else {
+                        hideEmptyResult()
+                    }
+                } else {
+                    showEmptyResult()
+                }
+            }
+        })
+    }
+
+    private fun startLoading() {
+        val query = search.text.toString()
+        hideEmptyResult()
+        search_loading.visibility = View.VISIBLE
+        var loadingText = getString(R.string.search_user_loading)
+        loadingText = loadingText.replace("%", "\"<b>$query</b>\"")
+        search_loading_label.text = HtmlCompat.fromHtml(loadingText,
+                HtmlCompat.FROM_HTML_MODE_COMPACT)
+        (search_loading_icon.drawable as AnimationDrawable).start()
+    }
+
+    private fun stopLoading() {
+        search_loading.visibility = View.GONE
+        (search_loading_icon.drawable as AnimationDrawable).start()
+    }
+
+    private fun showEmptyResult() {
+        search_user_empty_result.visibility = View.VISIBLE
+        var emptyResultText = getString(R.string.search_user_no_results)
+        emptyResultText += " \"<b>$query</b>\""
+        no_results_label.text = HtmlCompat.fromHtml(emptyResultText, HtmlCompat.FROM_HTML_MODE_COMPACT)
+    }
+
+    private fun hideEmptyResult() {
+        search_user_empty_result.visibility = View.GONE
+    }
+
+    private fun searchDashPayProfile() {
+        adapter.results = listOf()
+        hideEmptyResult()
+        if (this::searchDashPayProfileRunnable.isInitialized) {
+            handler.removeCallbacks(searchDashPayProfileRunnable)
+        }
+        if (query.length >= USERNAME_MIN_LENGTH) {
+            searchDashPayProfileRunnable = Runnable {
+                dashPayViewModel.searchUsernames(query)
```

Comment:
Since no value is specified for userId, this search will not retrieve any contact information for the current user with respect to the search results.

Of course, there won't any contact results to retrieve until this app can Request contacts.